### PR TITLE
PayloadProcessor: copy data buffer before passing to uploader

### PR DESCRIPTION
**Pull request by janpriessnitz**

### CODE REVIEW

**Comment by janpriessnitz** [comment]
@cernvm-bot cloudtest

**Comment by janpriessnitz** [comment]
@cernvm-bot cloudtest

File: cvmfs/receiver/payload_processor.cc [hunk 1]
```diff
@@ -165,8 +165,12 @@ void PayloadProcessor::ConsumerEventCallback(
   FileInfo& info = pending_files_[event.id];
 
   if (!info.skip) {
-    upload::AbstractUploader::UploadBuffer buf(event.buf_size, event.buf);
-    uploader_->ScheduleUpload(info.handle, buf);
+    void *buf_copied = smalloc(event.buf_size);
```
Comment (siscia) [inline cvmfs/receiver/payload_processor.cc:168]: isn't worth to check that `buf_copied != null` and if it is just panic.

Comment (jblomer) [reply cvmfs/receiver/payload_processor.cc:168]: Yep, that's what `smalloc` is doing on top of `malloc`: checks that the memory could be allocated (the "s" in `smalloc` was meant for "safe")

**Pull request merged**